lib/glific/templates/interactive_templates.ex

```diff
@@ -113,17 +113,77 @@ defmodule Glific.Templates.InteractiveTemplates do
   @spec create_interactive_template(map()) ::
           {:ok, InteractiveTemplate.t()} | {:error, Ecto.Changeset.t()}
   def create_interactive_template(attrs) do
-    case validate_interactive_content_length(attrs) do
+    case contains_markdown_syntax?(attrs) do
       :ok ->
-        %InteractiveTemplate{}
-        |> InteractiveTemplate.changeset(attrs)
-        |> Repo.insert()
+        case validate_interactive_content_length(attrs) do
+          :ok ->
+            %InteractiveTemplate{}
+            |> InteractiveTemplate.changeset(attrs)
+            |> Repo.insert()
+
+          {:error, message} ->
+            {:error, message}
+        end
 
       {:error, message} ->
         {:error, message}
     end
   end
 
+  @spec contains_markdown_syntax?(map()) :: :ok | {:error, String.t()}
+  defp contains_markdown_syntax?(attrs) do
+    case check_interactive_content(attrs) do
+      :ok -> :ok
+      {:error, message} -> {:error, message}
+    end
+  end
+
+  defp check_interactive_content(%{interactive_content: interactive_content}) do
+    check_options(interactive_content)
+  end
+
+  defp check_interactive_content(%{"interactive_content" => interactive_content}) do
+    check_options(interactive_content)
+  end
```

Comment:
Why do we have pattern matching same key in different format?, here the map in pattern match is same, the difference is key is atom and string in these functions. What can be the scenario for this to happen?

Reply:
I was thinking of a scenario when we  import the flow, but  its working fine with only atom, removed that